internal/ci/vulnerability_result_diff.go

```diff
@@ -26,7 +28,7 @@ func DiffVulnerabilityResults(oldRes, newRes models.VulnerabilityResults) models
 		resultPS := &result.Results[len(result.Results)-1]
 		for _, pv := range ps.Packages {
 			pkgs := oldRes.Results[sourceIdx].Packages
-			pkgIdx := slices.IndexFunc(pkgs, func(elem models.PackageVulns) bool { return elem.Package == pv.Package })
+			pkgIdx := slices.IndexFunc(pkgs, func(elem models.PackageVulns) bool { return reflect.DeepEqual(elem.Package, pv.Package) })
```

Comment:
Can we instead define a method on `models.PackageVulns` for this equality comparison? Is the main difference here that we want "groups" to be included in the comparison?

Reply:
Actually the code change here is not needed - I used to put "groups" in `PackageInfo` and that causes `==` not working. 
since I moved "groups" to `PackageVulns` so we can use `==` to compare `PackageInfo`.